dynamic_rest/processors.py

```diff
@@ -78,14 +78,18 @@ def process(self, obj, parent=None, parent_key=None, depth=0):
                     return
 
                 name = obj.serializer.get_plural_name()
-                pk = obj.instance.pk
+                pk = obj.pk_value or obj.instance.pk
+
+                # For polymorphic relations, `pk` can be a dict, so use the
+                # string representation (dict isn't hashable).
+                pk_key = repr(pk)
```

Comment:
Given equal values, seems like it, though this is far from exhaustive.

```python
>>> d = {}
>>> d['id'] = 'id'
>>> d['type'] = 'type'
>>> e = {}
>>> e['type'] = 'type'
>>> e['id'] = 'id'
>>> f = {'id':'id', 'type':'type'}
>>> repr(d)
"{'type': 'type', 'id': 'id'}"
>>> repr(e)
"{'type': 'type', 'id': 'id'}"
>>> repr(f)
"{'type': 'type', 'id': 'id'}"
```

One alternative would be to construct a tuple using the `id` and `type` but that would make assumptions about the structure/content of `pk`... Also, I think the only consequence of this hashing failing is that the same object might be sideloaded twice, which I don't think is too bad of a failure mode.